// react-login/src/Component/BgImage.js
import React, { Component, PropTypes } from 'react'
import { browserHistory, hashHistory } from 'react-router'
import { StyleSheet, css } from 'aphrodite';
import { Fonts, Colors } from '../themes'

class BgImage extends Component {
	static defaultProps = {
		//height: '3.14rem',
		height: '4rem',
		//position:'fixed'
	}
	render() {
		const { backgroundImage, wrap, title, text, share, onClick, height, position } = this.props
		const styles = StyleSheet.create({
			image: {
				backgroundImage: backgroundImage,
				//backgroundSize: '100% 100%',
				backgroundSize: 'cover',
				backgroundPosition: 'center center',
				position: position,
				height: height,
				textAlign: 'center',
				display: 'table',
				width: '100%',
				color: Colors.background,
				fontWeight: 700,
				boxSizing: 'border-box',
				zIndex: 20,
				left: 0,
				right: 0,
				margin: '0 auto'

			},
			wrap: {
				position: 'absolute',
				height: height,
				width: '100%',
				background: 'rgba(0,0,0,0.2)',
				display: wrap,
			},
			callBack: {
				height: '0.5rem',
				width: '0.5rem',
				lineHeight: '0.5rem',
				textAlign: 'center',
				position: 'absolute',
				top: '0.26rem',
				left: '0.26rem',
				backgroundColor: Colors.background,
				borderRadius: '50%',
				boxShadow: '0px 0px 20px rgba(0, 153, 255, 0.498039215686275)',
				zIndex: 2,
			},
			// share: {
			// 	height: '0.5rem',
			// 	width: '0.5rem',
			// 	lineHeight: '0.5rem',
			// 	textAlign: 'center',
			// 	position: 'absolute',
			// 	top: '0.26rem',
			// 	left: '1.38rem',
			// 	backgroundColor: 'rgba(0, 0, 0, 0.298039215686275)',
			// 	boxShadow: '0px 0px 20px rgba(0, 153, 255, 0.298039215686275)',
			// 	borderRadius: '50%',
			// 	zIndex: 2,
			// 	display: share
			// }
		})

		return (
			<div style={{ height: height }}>
				<div className={css(styles.image)}>
					<div className={css(styles.wrap)}></div>
					<div className={css(styles.callBack)}
						onClick={
							() => {
								browserHistory.goBack()
							}
						}
					>
						<img src={require('../../image/icon/u511.png')} style={{ width: '30%', verticalAlign: 'middle', }} />
					</div>
					<div style={{ display: 'table-cell', verticalAlign: 'middle', lineHeight: '1rem', position: 'relative', }}>
						<div style={{ fontSize: Fonts.title.fontSize }}>
							{title}
						</div>
						<div style={{ fontSize: Fonts.expert.fontSize }}>{text}</div>
					</div>
				</div>
			</div>
		)
	}
}


export default BgImage
