// react-login/src/Component/CarouselDemo.js
import React, { Component } from 'react';
import { browserHistory, hashHistory } from 'react-router'

import { Carousel } from 'antd-mobile';

import Course from './Course.js'
import { appService, appInfo } from '../service'
import { images } from '../themes'
const advertize = [
	{ id: 1, image: images.adv1, url: '/coursedetail/24/24/' },
	{ id: 2, image: images.adv2, url: '/activitydetail/22/' },
]


class CarouselDemo extends Component {
	constructor(props) {
		super(props)
		this.index = 0
		this.state = {
			data: [],
			isLoading: false
		}
	}

	componentWillMount() {
		this.fetchResources()
	}
  fetchResources(props) {
    let activityList = `${appInfo.address.activities}`
    if (this.props.name === 'activity') {
      appService.request(activityList)
        .then(result => {
            //let newArray = result.results.reverse().slice(0, 2)
            let newArray = result.results.slice(0, 2)
            console.log(result.results)
            this.setState({
              data: newArray,
              isLoading: true
            })
          }
        )
        .catch(error => {
            console.log(error)
          }
        )
    }
    else {
      this.setState({
        data: advertize,
        isLoading: true
      })
    }
  }

	render() {
		return (
			<div>
				{
					this.state.isLoading ?
						<Carousel
							className='my-carousel'
							autoplay={true}
							infinite
							dots={this.props.name==='activity'?true:false}
              //dots={false}
						>
							{
								this.state.data.map((item, index) => {
									return (
											<Course
												backgroundImage={'url(' + item.image + ')'}
												display='none'
												onClick={
													() => {
														let activityDetailUrl
													  if(this.props.name==='activity'){
															 activityDetailUrl = `/activitydetail/${item.id}/userId`
														}
													  else{
																//activityDetailUrl=`/advertisedetail/${item.id}/userId`
                                activityDetailUrl=item.url
														}
														browserHistory.push({
															pathname: activityDetailUrl
														})
                            /*
                            if(this.props.name==='activity'){
                              let activityDetailUrl = `/activitydetail/${item.id}/userId`
                              browserHistory.push({
                                pathname: activityDetailUrl
                              })
                            } else {
                              console.log('click')
                            }
                            */
													}
												}
											/>
									)
								})
							}
						</Carousel>
						:
						null
				}
			</div>
		);
	}
}


export default CarouselDemo;
