// react-login/src/Component/themes.js
import React, { Component } from 'react'
import { Fonts, Colors, images } from '../themes'
import { StyleSheet, css } from 'aphrodite'
class My extends Component {
	static defaultProps={
		rightContent:true
	}
	render() {
		const { title, display, onClick, rightContent,rightTitle,children} = this.props
	
		return (
			<div style={{backgroundColor:'#fff',position:'relative'}}>
				<div style={{ padding: '0.33rem 40px' }} onClick={onClick}>
					<span style={{ fontSize: '0.26rem', color: Colors.unActive, fontWeight: 500 }}>{title}</span>
					{children}
					{
						rightContent ?
							<span style={{ display: 'inline-Block', float: 'right', width: '0.18rem' }}>
								<img src={require('../../image/my/u2099.png')} width='100%' />
							</span>
							:
							<span  style={{ display: 'inline-Block', float: 'right',fontSize: '0.26rem', }} >
							<span style={{color:Colors.themeGreen,fontWeight:500}}>{rightTitle}</span>
							</span>

					}
				</div>
				<div style={{ display: !display ? 'none' : 'block' }}>
					<Line height='0.02rem' backgroundColor='#f9f9f9' />
				</div>
			</div>
		)
	}
}

class Line extends Component {
	render() {
		const { height, backgroundColor } = this.props
		const styles = StyleSheet.create({
			line: {
				backgroundColor: backgroundColor,
				height: height,
			}
		})
		return (
			<div className={css(styles.line)}></div>
		)
	}
}

My.propTypes = {
	rightContent: React.PropTypes.bool
};

export { My, Line }
