// react-login/src/themes/ApplicationStyles.js
import React, { Component } from 'react'
import Fonts from './Fonts'
import { StyleSheet, css } from 'aphrodite';
import Colors from './Colors'
import images from './Images.js'
const ApplicationStyles = {
	nav: {
		backgroundColor: '#fff',
		position: 'fixed',
		top: 0,
		color: '#000',
		fontWeight: 700,
		zIndex: 5,
		width: '100%',
		height: '0.9rem',
	},
	next: {
		width: '2.8rem',
		height: '0.8rem',
		borderRadius: '1.4rem',
		textAlign: 'center',
		color: Colors.navbj,
		border: '2px solid #fff',
		fontSize: '0.28rem',
		position:'fixed',
		left:0,
		right:0,
		margin:'0 auto',
		lineHeight: '0.8rem',
		fontWeight: 500,
		bottom:'0.2rem'

	},
	navBar: {
		color: Colors.navbj,
		fontSize: '0.28rem',
		backgroundColor: 'transparent',
	},
	backgroundImage: {
		backgroundSize: 'cover',
		backgroundPosition: 'center center',
		paddingBottom: '0.5rem',
	},
	pleaseChooseText: {
		marginTop: '0.47rem',
		textAlign: 'center',
		color: Colors.name,
		fontSize: '0.28rem',
		marginBottom: '0.35rem'
	},
	circle: {
		display: 'inline-Block',
		width: '1.58rem',
		height: "1.58rem",
		backgroundColor: 'rgba(0, 0, 0, 0.0980392156862745)',
		borderRadius: '50%',
		color: Colors.navbj,
		fontSize: '0.28rem',
		lineHeight: '1.58rem',
		textAlign: 'center',
	},
	circleActive: {
		display: 'inline-Block',
		width: '1.58rem',
		height: "1.58rem",
		backgroundColor: Colors.theme,
		borderRadius: '50%',
		color: Colors.navbj,
		fontSize: '0.28rem',
		lineHeight: '1.58rem',
		textAlign: 'center',
	},
	bottomText: {
		fontSize: Fonts.expert.fontSize,
		color: Colors.background,
		borderRadius: '0.12rem',
		margin: '0 auto',
		width: '3rem',
		padding: '0.15rem',
		border: '0.02rem solid rgba(51, 204, 102, 1)',
		marginTop: '0.2rem',
		marginBottom: '0.2rem',
		textAlign: 'center',
	},
	bottom: {
		fontSize: Fonts.expert.fontSize,
		color: Colors.themeGreen,
		borderRadius: '0.12rem',
		width: '3rem',
		padding: '0.15rem',
		border: '0.02rem solid rgb(0, 128, 0)',
		textAlign: 'center',
		margin: '0 auto',
		position: 'fixed',
		bottom: '0.2rem',
		left: 0,
		right: 0,
	},
	enroll: {
		width: '90%',
		height: '0.8rem',
		borderRadius: '1.4rem',
		textAlign: 'center',
		color: Colors.navbj,
		//backgroundColor: 'rgba(255, 204, 51, 1)',
    backgroundColor: Colors.theme,
		//border: '2px solid #fff',
		fontSize: '0.28rem',
		left: 0,
		right: 0,
		margin: '0 auto',
		lineHeight: '0.8rem',
		bottom: '0.2rem',
		position: 'fixed',
		fontWeight: 800,
	},
	login:{
		width: '90%',
		height: '0.8rem',
		borderRadius: '1.4rem',
		textAlign: 'center',
		color: Colors.navbj,
		backgroundColor: 'rgba(255, 204, 51, 1)',
		border: '2px solid #fff',
		fontSize: '0.28rem',
		margin: '0 auto',
		lineHeight: '0.8rem',
		marginbottom: '0.5rem',
		marginTop:'0.5rem',
		fontWeight: 800,
	},
	yellow: {
		width: '3rem',
		margin: '1rem auto',
		height: '3rem',
		backgroundImage: 'url(' + images.circle + ')',
		backgroundSize: 'cover',
		color: Colors.goodWorks,
		fontSize: Fonts.title.fontSize,
		display:'table',
		textAlign:'center',
		position:'relative'
	},

}

export default ApplicationStyles
