// react-login/src/themes/Images.js
const images = {
  ex5: require('../../image/home/u33_state0.jpeg'),
  //home: require('../../image/home/u1301.png'),
  home: require('../../image/home/01.jpg'),
  ex12: require('../../image/works/u1429.png'),
  ex14: require('../../image/activity/u274.JPG'),
  ex15: require('../../image/works/u240.png'),
  ex16: require('../../image/works/u248.png'),
  ex17: require('../../image/home/u57_state0.jpeg'),
  bg: require('../../image/coursecustomized/bj.jpg'),
  upLoadWorks: require('../../image/worksdetail/u382.png'),
  courseDetail: require('../../image/courseDetail/u104.jpg'),
  payCourseBg: require('../../image/payCourse/u210.jpg'),
  logo: require('../../image/payCourse/u226.png'),
  circle: require('../../image/payCourse/u190.png'),
  ele: require('../../image/home/u28.jpg'),
  //online: require('../../image/home/u1388_state0.jpg'),
  online: require('../../image/home/04.jpg'),
  //online1Img: require('../../image/home/u86_state0.jpg'),
  online1Img: require('../../image/home/02.jpg'),
  online2Img: require('../../image/home/u93_state0.jpg'),
  example1: require('../../image/home/u544.png'),
  example2: require('../../image/home/u554.png'),
  comment: {
    ex1: require('../../image/comment/u1574.png'),
    ex2: require('../../image/comment/u1585.png'),
  },
  enterBg: require('../../image/login/topimage.jpg'),
  man: require('../../image/coursecustomized/man.png'),
  girl: require('../../image/coursecustomized/girl.png'),
  light: require('../../image/coursecustomized/light.png'),
  p1: require('../../image/coursecustomized/01.png'),
  p2: require('../../image/coursecustomized/02.png'),
  p3: require('../../image/coursecustomized/03.png'),
  p4: require('../../image/coursecustomized/04.png'),
  p5: require('../../image/coursecustomized/05.png'),
  p6: require('../../image/coursecustomized/06.png'),
  p7: require('../../image/coursecustomized/07.png'),
  p8: require('../../image/coursecustomized/08.png'),
  age: require('../../image/coursecustomized/001.png'),
  line: require('../../image/coursecustomized/002.png'),
  //adv1:require('../../image/home/u544.png'),
  adv1:require('../../image/home/05.jpg'),
  //adv2:require('../../image/home/u554.png'),
  adv2:require('../../image/home/jdzyx.jpeg'),
  //adv3:require('../../image/home/u96.jpg'),
  adv3:require('../../image/home/05.jpg'),
}

export default images
