// react-login/src/service/index.js
import { request } from './WebService'
import  address from './Information'


var appService = {
  request,
}
var appInfo = {
  address,
}


// 全局基础数据及常用API
var appCore = {

  //课程的分类
  course: {
    category: {
      my: '我的定制课程',
      expert: '专家课程',
      improve: '系列提高课程',
      usual: '常规系列课程',
    },
  },
  home:{
    newActivity:'最新活动',
    works:'上传作品 免费点评',
    addCourse:'添加定制课程',
    moreCourse:'更多课程',
  },
  me:{
    myCourse:{
      text:'我的课程',
    },
    myWorks:{
      text:'我的作品',
    },
    growRecord:{
      text:'成长记录',
    },
    points:{
      text:'我的积分',
      balance:'积分余额',
      cash:'提现',
      exchangge:'兑换',
    },
    order:{
      text:'我的订单',
    },
    appoint:{
      text:'预约',
    },
    request:{
      text:'邀请',
    },
    introduce:{
      text:'转介绍',
    },
    card:{
      text:'会员卡绑定',
    }
  },
  target: {
    max: 5,
    min: 1,
    default: 2,
    toast: {
      maxToast: '请选择不超过5项',
      minToast: '请至少选择一项'
    },
  },
  study:{
    pleaseChoose:'请选择',
  },
  navTitle: {
    target: '提升目标',
    study:'以前学过',
    ability:'三商能力',
    character:'性格特征',
    performance:'生理心理在美术上的表现',
    courseCustomized:'基本资料',
    myWorks:'作品',
    myCourse:'我的课程',
    myPoints:'我的积分',
    myAppoints:'预约课程',
    pay:'支付',
    studentInfo:'基本信息',
    moreInfo:'更多信息',
    comment:'作品点评',
    baoMing:'报名信息'
  },

  next: {
    text: '下一步',
  },
  isLoading: false

}

export { appCore,appService,appInfo, }
