// react-login/src/service/Information.js
/**
 * Global app variables and infomation
 */
const serviceDomain = 'http://dongwu-inc.com:9999/'
//const serviceDomain = 'http://weixin.dongwu-inc.com:8000/'
const address = {
  domain: serviceDomain,
  onlinecourseSets:`${serviceDomain}api/online_course_sets/`,//线上课程列表
  offLineSets:`${serviceDomain}api/offline_course_sets/`,//线下课程列表
  schedule:`${serviceDomain}api/course_schedules/`,//获取课程表
  myOfflineList:`${serviceDomain}api/profile/offline_course_sets/`,//我的线下课程列表
  myOnlineList:`${serviceDomain}api/profile/online_course_sets/`,//我的线上课程列表
  myActivityList:`${serviceDomain}api/profile/activities/`,//我的活动列表
  courseCategory:`${serviceDomain}api/course_categories/`,//课程分类
  workCategory:`${serviceDomain}api/work_categories/`,//作品分类列表
  workList:`${serviceDomain}api/works/`,//作品列表
  workComments:`${serviceDomain}api/work_comments/`,
  blogList:`${serviceDomain}api/blogs/`,//圈子列表
  blogComments:`${serviceDomain}api/blog_comments/`,//圈子评论
  workLikes:`${serviceDomain}api/work_likes`,//作品点赞情况
  blogLikes:`${serviceDomain}api/blog_likes/`,//圈子点赞情况
  franchisees:`${serviceDomain}api/franchisees`,//代理点列表
  activities:`${serviceDomain}api/activities/` ,//活动列表
  example:`${serviceDomain}api/articles/`, //榜样的力量列表
  publishes:`${serviceDomain}api/published_course_sets/`, //获取加盟店课程接口
  courseList: `${serviceDomain}api/courses/`,
  activityEnroll:`${serviceDomain}api/activity_enrolls/?activity`,//报名接口
  releaseComments:`${serviceDomain}api/work_comments/`,//写评论
  courseSyllabuses: `${serviceDomain}api/syllabuses/`,
  workCategories: `${serviceDomain}api/work_categories/`,
  work: `${serviceDomain}api/works/`,
  register:`${serviceDomain}accounts/register/`,//注册
  login: `${serviceDomain}accounts/login/`,//登录
  logout:`${serviceDomain}accounts/logout/`,//退出登录
  customizeCourse:`${serviceDomain}api/customize_course_set/`,//定制课程
  me:`${serviceDomain}accounts/me/`,//我的
  myRelease:`${serviceDomain}api/profile/blogs/`,//我的发布
  changePassword:`${serviceDomain}accounts/change_password/`,//修改密码
  offlineEnroll:`${serviceDomain}api/course_enrolls/`,//线下课程报名
  sign:`${serviceDomain}api/sign/`,//签到
  joinStudy:`${serviceDomain}api/online_courseset_plans/`,//线上课程加入学习
  cityList:`${serviceDomain}accounts/available_cities/`,//开通城市列表
  myCoustomizeCourse:`${serviceDomain}api/profile/customize_course_sets/`,//我的定制课程列表
  myAppoint:`${serviceDomain}api/profile/course_reservations/`,//我的预约
  myOrder:`${serviceDomain}api/profile/orders/`,//我的订单
  scores:`${serviceDomain}api/profile/scores/`,//积分纪录
  outCourse:`${serviceDomain}api/online_course_sets/unenjoyed/`,//退出课程
  reservation:`${serviceDomain}api/course_reservations/`,//约课
  shareSuccess:`${serviceDomain}api/share_success/`, //分享成功
  workFees:`${serviceDomain}api/work_fees/`, //作品打赏
  blogFees:`${serviceDomain}api/blog_fees/`, //圈子打赏
  rankBorads:`${serviceDomain}api/rank_borads/`, //打榜
  rankBoradLikes:`${serviceDomain}api/rank_borad_likes/`, //打榜顶赞
  myRankBorads:`${serviceDomain}api/profile/rank_borads/`, //我的打榜
  myDeposits:`${serviceDomain}api/profile/deposits/`, //我的提现记录
  deposits:`${serviceDomain}api/deposits/`, //提现
  customizeCourseRecords:`${serviceDomain}api/customize_course_set_records/`, //我的定制课程记录
  courseStudyRecords:`${serviceDomain}api/course_study_records/`, //课程学习记录
  blogRecommend: `${serviceDomain}api/blog_recommendations/`, // 圈子推荐
  rankBoradRecommend: `${serviceDomain}api/rank_borad_recommendations/`, // 圈子推荐

  weixinUnifiedorder: `${serviceDomain}weixin/unifiedorder/`,//分享成功
  weixinAccessToken: `${serviceDomain}weixin/access_token/`, //分享成功
  weixinSignature: `${serviceDomain}weixin/signature/`//分享成功
}
export default address
